src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -52,33 +52,45 @@ public function validate($value, Constraint $constraint)
 
         $value = (string) $value;
 
-        if (null === $constraint->strict) {
-            $constraint->strict = $this->isStrict;
+        if (null === $constraint->mode) {
+            $constraint->mode = $this->mode;
         }
 
-        if ($constraint->strict) {
+        var_dump($constraint->mode);
+
+        if (Email::VALIDATION_MODE_RFC === $constraint->mode) {
             if (!class_exists('\Egulias\EmailValidator\EmailValidator')) {
-                throw new RuntimeException('Strict email validation requires egulias/email-validator ~1.2|~2.0');
+                throw new RuntimeException('RFC email validation requires egulias/email-validator ~1.2|~2.0');
             }
 
-            $strictValidator = new \Egulias\EmailValidator\EmailValidator();
+            $rfcValidator = new \Egulias\EmailValidator\EmailValidator();
 
-            if (interface_exists(EmailValidation::class) && !$strictValidator->isValid($value, new NoRFCWarningsValidation())) {
+            if (interface_exists(EmailValidation::class) && !$rfcValidator->isValid($value, new NoRFCWarningsValidation())) {
                 $this->context->buildViolation($constraint->message)
-                    ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Email::INVALID_FORMAT_ERROR)
-                    ->addViolation();
+                              ->setParameter('{{ value }}', $this->formatValue($value))
+                              ->setCode(Email::INVALID_FORMAT_ERROR)
+                              ->addViolation();
 
                 return;
-            } elseif (!interface_exists(EmailValidation::class) && !$strictValidator->isValid($value, false, true)) {
+            } elseif (!interface_exists(EmailValidation::class) && !$rfcValidator->isValid($value, false, true)) {
                 $this->context->buildViolation($constraint->message)
-                    ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Email::INVALID_FORMAT_ERROR)
-                    ->addViolation();
+                              ->setParameter('{{ value }}', $this->formatValue($value))
+                              ->setCode(Email::INVALID_FORMAT_ERROR)
+                              ->addViolation();
 
                 return;
             }
-        } elseif (!preg_match('/^.+\@\S+\.\S+$/', $value)) {
+        } elseif (Email::VALIDATION_MODE_LOOSE === $constraint->mode && preg_match('/^.+\@\S+\.\S+$/', $value)) {
```

Comment:
this is broken. You removed the `!`, so considering any email *with* a `@` in it as invalid.